File: ProjectEulerPython/problems/problem_025.py
from problems.problem import Problem

class Problem025(Problem):

  def calculate_answer(self) -> int:
    a = 1
    b = 1
    total_digits = 1000
    detail_size = total_digits // 10
    detail_threshold = detail_size

    answer = 3
    while True:
      c = a + b
      digit_count = len(str(c))
      if digit_count == total_digits:
        break
      elif digit_count >= detail_threshold:
        self.print_detail("index = " + str(answer) + "; length = " + str(digit_count))
        detail_threshold += detail_size
      a = b
      b = c
      answer += 1

    return answer

File: ProjectEuler/Problems/Problem025.cs
using Microsoft.Extensions.Logging;

namespace ProjectEuler.Problems
{
    public class Problem025 : Problem<long>
    {
        public Problem025(ILogger<Problem025> logger) : base(logger) { }

        protected override long CalculateAnswer()
        {
            var a = "1";
            var b = "1";
            var totalDigits = 1000;
            var detailSize = totalDigits / 10;
            var detailThreshold = detailSize;

            string c;
            var answer = 3L;
            while (true)
            {
                c = MathHelpers.AddIntegerStrings(a, b);
                if (c.Length == totalDigits) break;
                else if (c.Length >= detailThreshold)
                {
                    Logger.LogDebug("Index = {answer}; Length = {length}", answer, c.Length);
                    detailThreshold += detailSize;
                }
                a = b;
                b = c;
                ++answer;
            }

            return answer;
        }
    }
}
